### PR TITLE
Added ES6 Promises, upgraded Coffeescript

### PR DESCRIPTION
**Pull request by drhurd**
Per issue #17. Also updated Drone to auto publish (which we haven't done in a very long time).

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -2,9 +2,9 @@ test-cov:
 	rm -rf lib-js lib-js-cov
 	coffee -c -o lib-js lib
 	jscoverage lib-js lib-js-cov
-	NODE_ENV=test TEST_COV=1 node_modules/mocha/bin/mocha --compilers coffee:coffee-script -R html-cov test/get.coffee test/post.coffee | tee coverage.html
+	NODE_ENV=test TEST_COV=1 node_modules/mocha/bin/mocha --compilers coffee:coffee-script/register -R html-cov test/get.coffee test/post.coffee | tee coverage.html
 	open coverage.html
 test:
-	NODE_ENV=test node_modules/mocha/bin/mocha --bail --compilers coffee:coffee-script test/get.coffee test/post.coffee test/nock.coffee
+	NODE_ENV=test node_modules/mocha/bin/mocha --bail --compilers coffee:coffee-script/register test/get.coffee test/post.coffee test/nock.coffee
```
Comment (drhurd) [inline Makefile]: Upgrades the coffeescript version mocha uses to test.

File: lib/quest.coffee [hunk 2]
```diff
@@ -39,6 +40,13 @@ should_redirect = (options, resp) ->
   options.method not in ['PATCH', 'PUT', 'POST', 'DELETE']))
 
 quest = (options, cb) ->
+  if cb == undefined # making this explicit due to problems with coffeescript 1.9.3
```
Comment (templaedhel) [inline lib/quest.coffee]: What problems?

File: test/get.coffee [hunk 3]
```diff
@@ -304,3 +304,28 @@ describe 'quest', ->
         quest options, (err, resp, body) ->
           assert.ifError err
           done()
+
+      it 'is thenable on success', (done) ->
```
Comment (templaedhel) [inline test/get.coffee]: Thoughts on putting these inside their own describe? Like `describe "promise support" ->`? I could go either way but if it makes it more clear I think it makes sense?

File: test/get.coffee [hunk 4]
```diff
@@ -304,3 +304,28 @@ describe 'quest', ->
...
+        @timeout 20000
+        options =
+          uri: "#{protocol}://httpbin.org/get"
+        quest options
+          .then (resp) ->
```
Comment (templaedhel) [inline test/get.coffee]: How I'm reading this is that `then` is a function that takes in two arguments, both of which are functions as well. Is that correct?

**Comment by templaedhel** [comment]
@drhurd Left a few comments - in general the changes look good but I want to clarify a few things before LGTMing.

File: test/get.coffee (hunk 4 above)
Comment (drhurd) [inline test/get.coffee]: That is correct - for more info see [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise). Then() takes 2 functions - a success callback, and a failure callback.

File: test/get.coffee (hunk 3 above)
Comment (drhurd) [inline test/get.coffee]: Agreed, makes more sense to me as well.

File: lib/quest.coffee (hunk 2 above)
Comment (drhurd) [inline lib/quest.coffee]: After switching mocha to using coffeescript 1.9.3, I can't seem to reproduce the problem. Changing to normal coffeescript syntax.

File: lib/quest.coffee [hunk 5]
```diff
@@ -39,6 +40,13 @@ should_redirect = (options, resp) ->
   options.method not in ['PATCH', 'PUT', 'POST', 'DELETE']))
 
 quest = (options, cb) ->
+  if not cb
```
Comment (templaedhel) [inline lib/quest.coffee]: `unless`

**Comment by templaedhel** [comment]
LGTM - Lets merge. Feel free to rebase these commits together first though.

**Pull request merged**